cloud-aws/src/main/java/com/sequenceiq/cloudbreak/cloud/aws/AwsIDBrokerObjectStorageValidator.java

```diff
@@ -0,0 +1,258 @@
+import com.sequenceiq.common.api.cloudstorage.StorageLocationBase;
+import com.sequenceiq.common.model.CloudIdentityType;
+import com.sequenceiq.common.model.CloudStorageCdpService;
+
+public class AwsIDBrokerObjectStorageValidator {
+    private static final Logger LOGGER = LoggerFactory.getLogger(AwsIDBrokerObjectStorageValidator.class);
+
+    private final AmazonIdentityManagement iam;
+
+    public AwsIDBrokerObjectStorageValidator(AmazonIdentityManagement amazonIdentityManagement) {
+        this.iam = amazonIdentityManagement;
+    }
+
+    public void validateObjectStorage(List<CloudFileSystemView> cloudFileSystems) {
+        for (CloudFileSystemView cloudFileSystemView : cloudFileSystems) {
+            CloudS3View cloudFileSystem = (CloudS3View) cloudFileSystemView;
+            InstanceProfile instanceProfile = getInstanceProfile(cloudFileSystem.getInstanceProfile());
+            if (instanceProfile != null) {
+                CloudIdentityType cloudIdentityType = cloudFileSystem.getCloudIdentityType();
+                if (CloudIdentityType.ID_BROKER.equals(cloudIdentityType)) {
+                    checkInstanceProfileTrust(instanceProfile);
+                    checkIDBrokerAssumeRolePermissions(instanceProfile, cloudFileSystem);
+                    checkDataAccessRolePermissions(cloudFileSystem);
+                    checkRangerAuditRolePermissions(cloudFileSystem);
+                } else if (CloudIdentityType.LOG.equals(cloudIdentityType)) {
+                    checkInstanceProfileTrust(instanceProfile);
+                }
+            }
+        }
+    }
+
+    private InstanceProfile getInstanceProfile(String instanceProfileArn) {
+        InstanceProfile instanceProfile = null;
+        if (instanceProfileArn != null && instanceProfileArn.contains("/")) {
+            String instanceProfileName = instanceProfileArn.split("/", 2)[1];
+            GetInstanceProfileRequest instanceProfileRequest = new GetInstanceProfileRequest().withInstanceProfileName(instanceProfileName);
+            try {
+                instanceProfile = iam.getInstanceProfile(instanceProfileRequest).getInstanceProfile();
+            } catch (NoSuchEntityException | ServiceFailureException e) {
+                throw new CloudConnectorException(String.format("Unable to find instance profile: %s",
+                    instanceProfileArn), e);
+            }
+        }
+        return instanceProfile;
+    }
+
+    private void checkDataAccessRolePermissions(CloudS3View cloudFileSystem) {
+        // This list needs to match the IDBMMS list of data access users
+        // https://github.infra.cloudera.com/thunderhead/thunderhead/blob/master/
+        //  services/idbrokermappingmanagement/src/main/java/com/cloudera/thunderhead/
+        //  service/idbrokermappingmanagement/server/IdBrokerMappingManagementService.java#L109
+        List<String> dataAccessUsers = Arrays.asList("hbase", "hdfs", "hive", "impala", "yarn",
```

Comment:
maybe this list should be the same what we are using in AwsMockAccountMappingService?

Reply:
The list in `AwsMockAccountMappingService` is ALL the users which we don't want. We actually want to separate by data access and baseline/audit. The only place this logic exists today is in the IDBMMS service. The comment above tries to explain the difference.

The `MOCK_IDBROKER_USER_MAPPINGS` list in `AwsMockAccountMappingService` should be a superset of the `dataAccessUsers` list and the `rangerAuditUsers` list. Basically union the two lists together.

PS - I honestly wish the mock stuff was removed since it causes a disconnect with IDBMMS. @lajosrodek are there plans to remove the mock stuff? 

PPS - The more I deal w/ IDBMMS and Cloudbreak - it would be nice if IDBMMS wasn't a completely separate project in a separate repo :(

Reply:
Another thought is to set the data access and ranger audit users in `AwsMockAccountMappingService`. Then literally union them together to create the "mock" list. This would match a lot closer to what IDBMMS is doing.

Reply:
@risdenk I wish the mock impl was removed some day too. 😄 I think it is still useful (actually essential) for integration testing CB when IDBMMS is likely not running to have a complete E2E setup. @doktoric may confirm this.

Reply:
@risdenk Regarding the (overly broad) user list in `AwsMockAccountMappingService`: Agreed, it should be indeed simplified to match the current data access & ranger audit lists in IDBMMS. We chose to include all possible daemon users for laziness, and also because the authoritative list of IDBMMS had not yet been finalized by that time. Shall I raise a separate PR for that, or would you prefer covering that change here?
Fun fact: there is also `AzureMockAccountMappingService` containing yet another duplicate of the above jumbo user list. 😄 

CC @bergerdenes @pdarvasi

Reply:
Lets do a separate change since it affects AWS and Azure.

Reply:
Agreed. I just raised CB-4438 to track this.